torchx/lib/torchx/backend.ex

```diff
@@ -438,13 +441,26 @@ defmodule Torchx.Backend do
   ## Aggregators
 
   @impl true
-  def sum(%T{type: out_type} = out, %T{} = t, opts) do
-    check_type!(out_type)
+  def sum(%T{} = out, %T{} = t, opts) do
+    out =
```

Comment:
@josevalim I ended up having to make sum do the conversion automatically, because otherwise Nx.LinAlg.determinant would have to be aware of this quirk. WDYT about this? I think the same could be done to Nx.product.

Reply:
Also, it seems that `any?`, `all` and `product` all need the same "aggregate over axes" logic. Should I do this in this PR or should I leave it to a follow-up one?

Reply:
Also also, shouldn't we rename `any?` to `any`?